Fix 500 error when registry content is missing on the filesystem

```
Jan 26 16:39:55 rn000052275 gunicorn[238574]: pulp [-]:  - - [26/Jan/2022:22:39:55 +0000] "HEAD /v2/ee-supported-rhel8/blobs/sha256:1e6a9467750ba441584f9678e1537abd9e3b028f063da3994befd760452196f3 HTTP/1.0" 302 0 "-" "skopeo/1.4.2-dev"
Jan 26 16:39:55 rn000052275 gunicorn[238573]: [2022-01-26 22:39:55 +0000] [238656] [ERROR] Error handling request
Jan 26 16:39:55 rn000052275 gunicorn[238573]: Traceback (most recent call last):
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib64/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    resp = await self._request_handler(request)
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib64/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    resp = await handler(request)
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib64/python3.8/site-packages/aiohttp/web_middlewares.py", line 119, in impl
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    return await handler(request)
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib/python3.8/site-packages/pulpcore/content/authentication.py", line 35, in authenticate
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    return await handler(request)
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib/python3.8/site-packages/pulp_container/app/registry.py", line 269, in get_by_digest
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    return await Registry._dispatch(artifact.file, headers)
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib/python3.8/site-packages/pulp_container/app/registry.py", line 78, in _dispatch
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    full_headers["Content-Length"] = str(file.size)
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib/python3.8/site-packages/django/db/models/fields/files.py", line 71, in size
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    return self.storage.size(self.name)
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib/python3.8/site-packages/django/core/files/storage.py", line 334, in size
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    return os.path.getsize(self.path(name))
Jan 26 16:39:55 rn000052275 gunicorn[238573]:  File "/usr/lib64/python3.8/genericpath.py", line 50, in getsize
Jan 26 16:39:55 rn000052275 gunicorn[238573]:    return os.stat(filename).st_size
Jan 26 16:39:55 rn000052275 gunicorn[238573]: FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/pulp/media/artifact/1e/6a9467750ba441584f9678e1537abd9e3b028f063da3994befd760452196f3'
Jan 26 16:39:55 rn000052275 gunicorn[238573]: [26/Jan/2022:22:39:55 +0000] "HEAD /pulp/container/ee-supported-rhel8/blobs/sha256:1e6a9467750ba441584f9678e1537abd9e3b028f063da3994befd760452196f3?
```

This happened when a new pulp server was provisioned, pointed at an existing database. I was able to reproduce the error by just nuking things under `/var/lib/pulp/media/artifact/`.

--- a/CHANGES/555.bugfix
+++ b/CHANGES/555.bugfix
@@ -1,0 +1,1 @@
+Return a more concise message exception on 500 during image pull when content is missing on the FS.

--- a/pulp_container/app/registry.py
+++ b/pulp_container/app/registry.py
@@ -1,5 +1,6 @@
 import logging
 import os
+from gettext import gettext as _
 
 from asgiref.sync import sync_to_async
 
@@ -64,17 +65,15 @@ class Registry(Handler):
             The :class:`aiohttp.web.FileResponse` for the file.
 
         """
+        path = os.path.join(settings.MEDIA_ROOT, file.name)
+        if not os.path.exists(path):
+            raise Exception(_("Expected path '{}' is not found").format(path))
+
         full_headers = MultiDict()
 
         full_headers["Content-Type"] = headers["Content-Type"]
         full_headers["Docker-Content-Digest"] = headers["Docker-Content-Digest"]
         full_headers["Docker-Distribution-API-Version"] = "registry/2.0"
-        full_headers["Content-Length"] = str(file.size)
-        full_headers["Content-Disposition"] = "attachment; filename={n}".format(
-            n=os.path.basename(file.name)
-        )
-
-        path = os.path.join(settings.MEDIA_ROOT, file.name)
         file_response = web.FileResponse(path, headers=full_headers)
         return file_response
 
@@ -123,7 +122,7 @@ class Registry(Handler):
             and tag.tagged_manifest.media_type not in accepted_media_types
         ):
             log.warn(
-                "OCI format found, but the client only accepts {accepted_media_types}.".format(
+                _("OCI format found, but the client only accepts {accepted_media_types}.").format(
                     accepted_media_types=accepted_media_types
                 )
             )

--- a/pulp_container/app/schema_convert.py
+++ b/pulp_container/app/schema_convert.py
@@ -6,6 +6,9 @@ import hashlib
 import itertools
 import json
 import logging
+
+from gettext import gettext as _
+
 from collections import namedtuple
 from jwkest import jws, jwk, ecc
 
@@ -323,4 +326,11 @@ def _get_manifest_dict(manifest):
 
 
 def _get_dict(artifact):
-    return json.load(artifact.file)
+    try:
+        return json.load(artifact.file)
+    except FileNotFoundError:
+        raise Exception(
+            _(
+                "Expected manifest file 'sha256:{}' needed for schema conversion is not found"
+            ).format(artifact.sha256)
+        )

